test/pgtest/copy-from-csv.pt

```diff
@@ -0,0 +1,139 @@
+# Note that this file will not work on PG due to the lack of "strict" parsing
+# from the Rust CSV crate: https://github.com/BurntSushi/rust-csv/issues/77
+
+send
+Query {"query": "DROP TABLE IF EXISTS t"}
+----
+
+until ignore=NoticeResponse
+ReadyForQuery
+----
+CommandComplete {"tag":"DROP TABLE"}
+ReadyForQuery {"status":"I"}
+
+send
+Query {"query": "CREATE TABLE t (i INT8, t TEXT)"}
+----
+
+until
+ReadyForQuery
+----
+CommandComplete {"tag":"CREATE TABLE"}
+ReadyForQuery {"status":"I"}
+
+# Note that we cannot differentiate between empty string and the quoted empty
+# string, both of which will generate NULL in MZ.
+send
+Query {"query": "COPY t FROM STDIN WITH (FORMAT CSV)"}
+CopyData "1,one\n"
+CopyData "2,\"two\"\n"
+CopyData "3,\"\"\"three\"\"\"\n"
+CopyData "4,\"fo,ur\"\n"
+CopyData "5,\"\"\n"
+CopyData "6,\n"
+CopyData "7,\"seven\n"
+CopyData "8,eight\n"
+CopyDone
+Query {"query": "SELECT * FROM t ORDER BY i"}
+----
+
+until
+ReadyForQuery
+ReadyForQuery
+----
+CopyIn {"format":"text","column_formats":["text","text"]}
+CommandComplete {"tag":"COPY 7"}
+ReadyForQuery {"status":"I"}
+RowDescription {"fields":[{"name":"i"},{"name":"t"}]}
+DataRow {"fields":["1","one"]}
+DataRow {"fields":["2","two"]}
+DataRow {"fields":["3","\"three\""]}
+DataRow {"fields":["4","fo,ur"]}
+DataRow {"fields":["5","NULL"]}
+DataRow {"fields":["6","NULL"]}
+DataRow {"fields":["7","seven\n8,eight\n"]}
+CommandComplete {"tag":"SELECT 7"}
+ReadyForQuery {"status":"I"}
+
+# Change options. Note that:
+# - After receiving terminating char, no other data should be accepted
+# - Quoted end of copy markers are still processed as end of copy markers
+send
+Query {"query": "DELETE FROM t"}
+Query {"query": "COPY t FROM STDIN WITH (FORMAT CSV, QUOTE '|', ESCAPE '#', NULL 'NS', HEADER true)"}
```

Comment:
Can you add tests with various combinations of delimiter, quote, and escape all being the same character?

Reply:
Such a good call. Found out that the delimiter and quote character must differ. Added a test where delimiter and escape are the same character, though.